src/test/java/net/snowflake/client/jdbc/ResultJsonParserV2Test.java

```diff
@@ -223,4 +229,41 @@ public static String stringToHex(String input) {
     }
     return sb.toString().toUpperCase();
   }
+
+  @Test
+  public void testObjectMapperWithLargeJsonString() throws Exception {
+    try {
+      // test with base64 encoded 16M string
+      ObjectMapper mapper = ObjectMapperFactory.getObjectMapper();
+      JsonNode jsonNode = mapper.readTree(generateBase64EncodedJsonString(16 * 1024 * 1024));
+      Assert.assertNotNull(jsonNode);
+      jsonNode = null;
+
+      // test with base64 encoded 32M string
+      ObjectMapperFactory.setMaxJsonStringLength(45000000);
+      mapper = ObjectMapperFactory.getObjectMapper();
+      jsonNode = mapper.readTree(generateBase64EncodedJsonString(32 * 1024 * 1024));
+      Assert.assertNotNull(jsonNode);
+    } catch (Exception ex) {
+      fail("failed to parse the large JSON string. err: " + ex.getMessage());
+    }
+  }
+
+  private String generateBase64EncodedJsonString(int numChar) {
+    StringBuilder sb = new StringBuilder();
+    StringBuilder a = new StringBuilder();
+
+    // create string with the specified size
+    for (int i = 0; i < numChar; i++) {
```

Comment:
can we use `SnowflakeUtil.randomAlphaNumeric` ?

Reply:
Updated